src/query/ts/m3db/types.go

```diff
@@ -78,6 +78,10 @@ type Options interface {
 	SetIteratorBatchingFn(IteratorBatchingFn) Options
 	// IteratorBatchingFn returns the batching function for the converter.
 	IteratorBatchingFn() IteratorBatchingFn
+	// SetInstrumented marks if the encoding step should have instrumentation enabled.
+	SetInstrumented(bool) Options
+	// Instrumented returns if the encoding step should have instrumentation enabled.
+	Instrumented() bool
```

Comment:
nit: this may be more useful if it had an `instrument.Options` on it, and the instrument options had a `TraceEnabled` method?

Reply:
Decided to hold off on this til we have reason to expand this beyond tracing